{{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/download.py

```diff
@@ -31,64 +33,51 @@ class FileConfig(BaseModel):
     )
 
 
-@dataclass
-class DemoFileStrategy:
-    """Strategy for retrieving data via local file."""
-
-    download_config: "ResourceConfig"
-
-    def initialize(
-        self, session: "Optional[Dict[str, Any]]" = None
-    ) -> "Dict[str, Any]":
-        """Initialize strategy.
+class SessionUpdateFile(SessionUpdate):
+    """Class for returning values from Download File strategy."""
 
-        This method will be called through the `/initialize` endpoint of the OTE-API
-        Services.
+    key: str = Field(..., description="Key to access the data in the cache.")
 
-        Parameters:
-            session: A session-specific dictionary context.
 
-        Returns:
-            Dictionary of key/value-pairs to be stored in the sessions-specific
-            dictionary context.
+@dataclass
+class DemoFileStrategy:
+    """Demo strategy for retrieving data from a local file.
 
-        """
-        return {}
+    **Registers strategies**:
 
-    def get(self, session: "Optional[Dict[str, Any]]" = None) -> "Dict[str, Any]":
-        """Execute the strategy.
+    - `("scheme", "fileDEMO")`
 
-        This method will be called through the strategy-specific endpoint of the
-        OTE-API Services.
+    """
 
-        Parameters:
-            session: A session-specific dictionary context.
+    download_config: "ResourceConfig"
 
-        Returns:
-            Dictionary of key/value-pairs to be stored in the sessions-specific
-            dictionary context.
+    def initialize(self, session: "Optional[Dict[str, Any]]" = None) -> SessionUpdate:
+        """Initialize."""
+        return SessionUpdate()
 
-        """
+    def get(self, session: "Optional[Dict[str, Any]]" = None) -> SessionUpdateFile:
+        """Read local file."""
         if (
             self.download_config.downloadUrl is None
             or self.download_config.downloadUrl.scheme != "file"
         ):
             raise ValueError(
                 "Expected 'downloadUrl' to have scheme 'file' in the configuration."
             )
-        filename = Path(self.download_config.downloadUrl.host).resolve()
+
+        filename = Path(self.download_config.downloadUrl.path).resolve()
+        if isinstance(filename, PosixPath):
+            filename = Path("/" + self.download_config.downloadUrl.host + str(filename))
```

Comment:
This can be done in a cleaner way. See your latest update of the oteapi-core file strategy.

Reply:
Indeed. This is taken from the oteapi-core `master` as I didn't want to implement stuff based on as yet un-merged stuff.

Reply:
Indeed, when EMMC-ASBL/oteapi-core#81 is merged, the special utility function can be used here instead.

Reply:
It has been merged and can now me used - this should be updated accordingly.